java/arcs/android/storage/service/IDevToolsProxy.aidl

```diff
@@ -0,0 +1,13 @@
+package arcs.android.storage.service;
+
+import arcs.android.storage.service.IStorageServiceCallback;
+
+/**
+ * Exposed API to manage storage for DevTools
+ */
+interface IDevToolsProxy {
+
+    void onBindingContextProxyMessage(in byte[] proxyMessage);
```

Comment:
I don't think this belongs here.

This interface should described the communication contract between the StorageService (concretely DevToolsProxyImpl) and the DevToolsService. This contract is: DevToolsService registers a callback with registerBindingContextProxyMessageCallback, and Storage service replies with the provided IStorageServiceCallback.

onBindingContextProxyMessage is a method on DevToolsProxyImpl (should be extracted to its own interface as I argue in other comment), and it's part of the interface between the storage stack and DevToolsProxy.

I can see at least 3 interfaces here between different layers:

StorageStack <-(1)-Kotlin-Interface-> DevToolsProxy <-(2)-AIDL-> DevToolsService <-(3)-WebSocket-> DevTools Frontend.

onBindingContextProxyMessage is part of interface (1) - it's also likely temporary and will be replaced by storage stack hooks, not binding context hooks. By putting this method in this .aidl you indicate it is a part of interface (2), which is not true.

Reply:
Removed onBindingContextProxyMessage from the aidl as also suggest by Jason.

Reply:
You may be blending 2 concepts together. There are these 3 interfaces I outline above, and they will all will be passing similar messages, yet they are different in implementation and have different constraints.

1) DevToolsProxy (I believe that's the name we discussed and you're proposing here) - the interface that will be available to the Storage Stack. This will need to be in `arcs/core`, cannot be tied to android. There will be an interface and a decoupled implementation that is backed by Android .aidl based communication.

2) IDevToolsStorageManager (I think that's the right object for this interface) - This is a way for DevToolsService to talk to the Storage Service. This is defined with AIDL.

3) WebSocket - here we pass (most likely) JSON between DevToolsService and some developer tool, e.g. a web page.

You may be blending layers a bit, because you added DevToolsProxy to AIDL and it seems like you plan to pass down the AIDL stub down the storage stack - which you cannot do without a layer of indirection (as core cannot be android specific).

I would ask you whether DevToolsProxy should not be pure interface, and whether the content of IDevToolsProxy should not be inlined in IDevToolsStorageManager instead.

You will need to resolve all these questions at the next PR when you try to pass down the devtools proxy down the storage stack - just pointing it out early.